src/monodroid/jni/android-system.h

```diff
@@ -26,6 +26,7 @@ namespace xamarin { namespace android { namespace internal
 		bool        DSOInApk = false;
 		MonoAotMode MonoAOT = MonoAotMode::MONO_AOT_MODE_NONE;
 		bool        MonoLLVM = false;
+		bool        EnableAssemblyPreload = false;
```

Comment:
@jonathanpeppers wonders if this logic should be inverted: if the property *isn't* set at all, we should default to preloading assemblies.

Which suggests that the default value should be `true`...

Reply:
Yeah my thought behind that was: new behavior/feature = new flag.

If we introduce `mono.enable_assembly_preload=1` and `$(AndroidEnablePreloadAssemblies)=True`, we are adding an `AndroidEnvironment` file that will now get generated on every project. It seems better for the build to only generate a file, when they opt into the new behavior. We can revisit this when the "new behavior" becomes the "default behavior".

Reply:
So a concern here came up on slack: https://xamarinhq.slack.com/archives/C03CEGRUW/p1549989040250300

I think the delay loading behavior is breaking the Xamarin.Forms previewer.  I definitely think we need to reverse the behavior where no env var is needed to get back to the old behavior. Unless we want to follow all the way through and test the previewer works with the `AndroidEnvironment` file on disk...

Reply:
Ok, after latest changes, this looks good now.